server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java

```diff
@@ -769,4 +774,83 @@ public XContentBuilder toXContent(final XContentBuilder builder, final Params pa
 
     }
 
+    static class IndexPressureStats implements ToXContentFragment {
+
+        private static final String COMBINED = "combined_coordinating_and_primary";
+        private static final String COMBINED_IN_BYTES = "combined_coordinating_and_primary_in_bytes";
+        private static final String COORDINATING = "coordinating";
+        private static final String COORDINATING_IN_BYTES = "coordinating_in_bytes";
+        private static final String PRIMARY = "primary";
+        private static final String PRIMARY_IN_BYTES = "primary_in_bytes";
+        private static final String REPLICA = "replica";
+        private static final String REPLICA_IN_BYTES = "replica_in_bytes";
+        private static final String ALL = "all";
+        private static final String ALL_IN_BYTES = "all_in_bytes";
+        private static final String COORDINATING_REJECTIONS = "coordinating_rejections";
+        private static final String PRIMARY_REJECTIONS = "primary_rejections";
+        private static final String REPLICA_REJECTIONS = "replica_rejections";
+        private static final String LIMIT = "limit";
+        private static final String LIMIT_IN_BYTES = "limit_in_bytes";
+
+        final long[] stats = new long[12];
```

Comment:
This approach means that we have to keep the numbers in the constructor aligned with the ones in `toXContent`, which is the sort of thing that tends to go wrong in future changes. Is it possible just to re-use `org.elasticsearch.index.stats.IndexingPressureStats` here? That'd save worrying about keeping everything lined up between the different representations. If that's not possible for some reason then I'd prefer each stat to be in its own named field.

Reply:
I reuse the object IndexingPressureStats in the function that sum values, say me if it's good. It's way clear and simple.